File: discover_javascript/manipulate_the_data.js
var https = require('https');

var options = {
    hostname: 'api.github.com',
    path: '/search/repositories?q=language:javascript&sort=stars&order=desc',
    headers: {
	'User-Agent': 'Holberton_School',
	'Authorization': 'token ' + process.env.TOKEN
    }
}

var combine = function streamToString(stream, cb) {
    const chunks = [];
    stream.on('data', (chunk) => {
	    chunks.push(chunk);
	});
    stream.on('end', () => {
	    cb(chunks.join(''));
	});
}

var req = https.request(options, function(res){
  combine(res,function(jsonString){
    var data = JSON.parse(jsonString);
    var array = data.items.map(function(project){
      return project.full_name;
    });
    console.log(array.join("\n"));
  });
});
req.end();

File: discover_ruby/manipulate_the_data.rb
require 'httpclient'
require 'json'

extheaders = {
  'User-Agent' => 'Holberton_School',
  'Authorization' => 'token 4d55fa9beb614fd08ed010f0442c5a7239e734ee'
}

uri = "https://api.github.com/search/repositories?q=language:ruby&sort=stars&order=desc"

clnt = HTTPClient.new
json_string = clnt.get_content(uri, extheaders)
res = JSON.parse(json_string)['items']

names = res.map {|e| e["full_name"]}
puts names.join("\n")
